Teleport tsh ssh crashing with nil pointer / invalid memory address SIGSEGV

Expected behavior:

```bash
tsh ssh -L 3306:mydatabase:3306 -N --proxy=secure.host.com root@bastion-host-1
```

I would expect this command to work as it did last week before I upgraded to *12.2.1-1*.

Current behavior:

Normally I proxy to my database using the `tsh ssh` command, but after upgrading from version *12.1.5-1* to version *12.2.1-1*, I am getting the following crash:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x35e404a]

goroutine 1 [running]:
github.com/gravitational/teleport/lib/client.(*TeleportClient).getSSHLoginFunc(0xc0004a0e00, 0xc000554000)
        github.com/gravitational/teleport/lib/client/api.go:3189 +0x12a
github.com/gravitational/teleport/lib/client.(*TeleportClient).Login(0xc0004a0e00, {0x4ef3780, 0xc000e6a360})
        github.com/gravitational/teleport/lib/client/api.go:3089 +0x1b8
github.com/gravitational/teleport/lib/client.RetryWithRelogin({0x4ef3780, 0xc000e6a360}, 0xc0004a0e00, 0xc00141e960)
        github.com/gravitational/teleport/lib/client/api.go:543 +0x205
main.onSSH.func1()
        github.com/gravitational/teleport/tool/tsh/tsh.go:3027 +0x90
main.retryWithAccessRequest(0xc000178e00, 0xc0004a0e00, 0xc00141eb10)
        github.com/gravitational/teleport/tool/tsh/tsh.go:2936 +0x66
main.onSSH(0xc000178e00)
        github.com/gravitational/teleport/tool/tsh/tsh.go:3026 +0xbd
main.Run({0x4ef4ee8, 0xc000e819c0}, {0xc000074080?, 0x6, 0x6}, {0x0, 0x0, 0x1613105?})
        github.com/gravitational/teleport/tool/tsh/tsh.go:1116 +0xbf25
main.main()
        github.com/gravitational/teleport/tool/tsh/tsh.go:499 +0x305
```

Bug details:
- Teleport version 12.2.1-1 (issue did not present in 12.1.5-1, so I have reverted to that for now)
- Recreation steps Try to proxy a port locally using the ssh command
- Debug logs:

```
2023-04-08T21:53:16+01:00 INFO [CLIENT]    ALPN connection upgrade required for "<secure bastion>:443": false. client/api.go:629
2023-04-08T21:53:16+01:00 DEBU [CLIENT]    Skipping connection to the local ssh-agent. client/keyagent.go:136
2023-04-08T21:53:16+01:00 WARN [CLIENT]    "Certificate was not loaded into agent because the agent at SSH_AUTH_SOCK does not appear\nto support SSH certificates. To force load the certificate into the running agent, use\nthe --add-keys-to-agent=yes flag." client/keyagent.go:139
2023-04-08T21:53:16+01:00 DEBU [KEYSTORE]  Reading certificates from path "/home/<user>/.tsh/keys/<secure bastion>/<user-email>-ssh/<secure bastion>-cert.pub". client/keystore.go:341
2023-04-08T21:53:16+01:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-04-07 07:30:48 +0000 UTC". client/client_store.go:91
2023-04-08T21:53:16+01:00 INFO [KEYAGENT]  Loading SSH key for user "<user-email>" and cluster "<secure bastion>". client/keyagent.go:195
2023-04-08T21:53:16+01:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-04-07 07:30:48 +0000 UTC". client/client_store.go:91
2023-04-08T21:53:16+01:00 DEBU [CLIENT]    Activating relogin on error ssh: cert has expired. client/api.go:530
2023-04-08T21:53:16+01:00 DEBU [CLIENT]    not using loopback pool for remote proxy addr: <secure bastion>:443 client/api.go:3950
2023-04-08T21:53:16+01:00 DEBU             Attempting GET <secure bastion>:443/webapi/ping/local webclient/webclient.go:129
2023-04-08T21:53:18+01:00 DEBU             ALPN connection upgrade required for "<secure bastion>:443": false. alpnproxy/conn_upgrade.go:81
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x35e404a]

goroutine 1 [running]:
github.com/gravitational/teleport/lib/client.(*TeleportClient).getSSHLoginFunc(0xc00029e380, 0xc00054a000)
        github.com/gravitational/teleport/lib/client/api.go:3189 +0x12a
github.com/gravitational/teleport/lib/client.(*TeleportClient).Login(0xc00029e380, {0x4ef3780, 0xc0001502d0})
        github.com/gravitational/teleport/lib/client/api.go:3089 +0x1b8
github.com/gravitational/teleport/lib/client.RetryWithRelogin({0x4ef3780, 0xc0001502d0}, 0xc00029e380, 0xc0014d4960)
        github.com/gravitational/teleport/lib/client/api.go:543 +0x205
main.onSSH.func1()
        github.com/gravitational/teleport/tool/tsh/tsh.go:3027 +0x90
main.retryWithAccessRequest(0xc00097a700, 0xc00029e380, 0xc0014d4b10)
        github.com/gravitational/teleport/tool/tsh/tsh.go:2936 +0x66
main.onSSH(0xc00097a700)
        github.com/gravitational/teleport/tool/tsh/tsh.go:3026 +0xbd
main.Run({0x4ef4ee8, 0xc000ba5400}, {0xc000074110?, 0x7, 0x7}, {0x0, 0x0, 0x1613105?})
        github.com/gravitational/teleport/tool/tsh/tsh.go:1116 +0xbf25
main.main()
        github.com/gravitational/teleport/tool/tsh/tsh.go:499 +0x305
```

OS details: `Linux <host> 6.2.9-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 30 Mar 2023 14:51:14 +0000 x86_64 GNU/Linux`

--- a/lib/client/api.go
+++ b/lib/client/api.go
@@ -3336,7 +3336,11 @@ func (tc *TeleportClient) DeviceLogin(ctx context.Context, certs *devicepb.UserC
 func (tc *TeleportClient) getSSHLoginFunc(pr *webclient.PingResponse) (SSHLoginFunc, error) {
 	switch pr.Auth.Type {
 	case constants.Local:
-		switch pr.Auth.Local.Name {
+		authType := constants.LocalConnector
+		if pr.Auth.Local != nil {
+			authType = pr.Auth.Local.Name
+		}
+		switch authType {
 		case constants.PasswordlessConnector:
 			// Sanity check settings.
 			if !pr.Auth.AllowPasswordless {

